stripe-ui-core/src/main/java/com/stripe/android/uicore/utils/StateFlowsComposeUtils.kt

```diff
@@ -0,0 +1,57 @@
+package com.stripe.android.uicore.utils
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.ProduceStateScope
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+private class DefaultProduceStateScope<T>(
+    state: MutableState<T>,
+    override val coroutineContext: CoroutineContext
+) : ProduceStateScope<T>, MutableState<T> by state {
+
+    override suspend fun awaitDispose(onDispose: () -> Unit): Nothing {
+        try {
+            suspendCancellableCoroutine<Nothing> { }
+        } finally {
+            onDispose()
+        }
+    }
+}
+
+@Composable
+private fun <T> produceState(
+    produceInitialValue: () -> T,
+    key: Any?,
+    producer: suspend ProduceStateScope<T>.() -> Unit
+): State<T> {
+    val result = remember { mutableStateOf(produceInitialValue()) }
+    LaunchedEffect(key) {
+        DefaultProduceStateScope(result, coroutineContext).producer()
+    }
+    return result
+}
+
+@Composable
+fun <T> StateFlow<T>.collectAsStateSafely(
+    context: CoroutineContext = EmptyCoroutineContext
+): State<T> = produceState(
+    produceInitialValue = remember { { value } },
```

Comment:
It seems like we should upstream this behavior into the `FlowToStateFlow` class. Then we get the benefit everywhere, not just compose. Is there any reason we can't?

Ideally we'd transform initially, then we'd update the value every time we emit a value in `FlowToStateFlow`.

Reply:
Maybe something like this: https://github.com/stripe/stripe-android/pull/8478

Reply:
Hmm, my idea might not work. If a mutable state flow updates it's internal `value`, without someone consuming it, then the value won't be update. :/